src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -223,16 +221,61 @@ else if (planKey.endsWith("-" + BuildPlanType.SOLUTION.getName())) {
         Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, exercise, null);
 
-        List<Result> results = new ArrayList<>();
-        var isExamMode = exercise.isExamExercise();
-
-        List<StudentParticipation> participations;
-        if (isExamMode) {
+        final List<StudentParticipation> participations;
+        if (exercise.isExamExercise()) {
             participations = studentParticipationRepository.findByExerciseIdWithEagerSubmissionsResultAssessorIgnoreTestRuns(exerciseId);
         }
         else {
             participations = studentParticipationRepository.findByExerciseIdWithEagerSubmissionsResultAssessor(exerciseId);
         }
+
+        List<Result> results = resultsForExercise(exercise, participations, withSubmissions);
+        log.info("getResultsForExercise took {}ms for {} results.", System.currentTimeMillis() - start, results.size());
+
+        return ResponseEntity.ok().body(results);
+    }
+
+    /**
+     * GET /exercises/:exerciseId/results-with-points-per-criterion : get the successful results for an exercise, ordered ascending by build completion date.
+     * Also contains for each result the points the student achieved with manual feedback. Those points are grouped as sum for each grading criterion.
+     *
+     * @param exerciseId of the exercise for which to retrieve the results.
+     * @param withSubmissions defines if submissions are loaded from the database for the results.
+     * @return the ResponseEntity with status 200 (OK) and the list of results with points in body.
+     */
+    @GetMapping("exercises/{exerciseId}/results-with-points-per-criterion")
+    @PreAuthorize("hasRole('TA')")
```

Comment:
should this really be possible for TAs? I guess this violates the double blind rules and should probably only allowed for Editors or even instructors

Reply:
Tutors were actually able to see the Scores button on the `/course-management/$ID/exercises` overview page. In commit 837ce2719e44a1a5f502abf2d6a300d639f033ef I updated it so that this button is only visible to instructors for all exercise types and also updated the endpoint permissions.